packages/text-input/src/TextInput.tsx

```diff
@@ -0,0 +1,331 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { css, cx } from '@leafygreen-ui/emotion';
+import Icon from '@leafygreen-ui/icon';
+import { uiColors } from '@leafygreen-ui/palette';
+import { useUsingKeyboardContext } from '@leafygreen-ui/leafygreen-provider';
+import { createDataProp } from '@leafygreen-ui/lib';
+
+export const State = {
+  None: 'none',
+  Valid: 'valid',
+  Error: 'error',
+} as const;
+
+export type State = typeof State[keyof typeof State];
+
+interface TextInputProps {
+  /**
+   * Text shown in bold above the input element.
+   */
+  label?: string;
+  /**
+   * Text that gives more detail about the requirements for the input.
+   */
+  description?: string;
+  /**
+   * Whether or not the field is optional.
+   * Default: false
+   */
+  optional?: boolean;
+  /**
+   * Whether or not the field is currently disabled.
+   * Default: false
+   */
+  disabled?: boolean;
+  /**
+   * Callback to be executed when the value of the input field changes.
+   */
+  onChange?: Function;
+  /**
+   * The placeholder text shown in the input field before the user begins typing.
+   */
+  placeholder?: string;
+  /**
+   * The message shown below the input field if the value is invalid.
+   */
+  errorMessage?: string;
+  /**
+   * The current state of the TextInput. This can be none, valid, or error.
+   */
+  state?: State;
+  /**
+   * The current value of the input field. If a value is passed to this prop, component will be controlled by consumer.
+   */
+  value?: string;
+  /**
+   * Callback used to set the value of the input field.
+   */
+  setValue?: Function;
```

Comment:
This shouldn't exist. If it's controlled then you just set the `value={value}` prop. There's no setter, we just all onChange when the value is changing.

Reply:
Can you explain further? I just kind of modeled my code after other hybrid controlled/uncontrolled components so maybe I'm missing something.

Reply:
Never mind, I figured it out! setValue prop has been removed.